verifier-py/insights_ansible_playbook_verifier/crypto.py

```diff
@@ -0,0 +1,252 @@
+import dataclasses
+import errno
+import logging
+import os
+import os.path
+import pathlib
+import shutil
+import tempfile
+import subprocess
+import typing
+
+logger = logging.getLogger(__name__)
+
+
+if os.geteuid() == 0 and os.path.isdir("/var/lib/insights-ansible-playbook-verifier/"):
+    TEMPORARY_GPG_HOME_PARENT_DIRECTORY = "/var/lib/insights-ansible-playbook-verifier/"
+    TEMPORARY_GPG_HOME_PARENT_DIRECTORY_PREFIX = "gpg-"
+else:
+    TEMPORARY_GPG_HOME_PARENT_DIRECTORY = "/tmp/"
+    TEMPORARY_GPG_HOME_PARENT_DIRECTORY_PREFIX = "insights-ansible-playbook-verifier-gpg-"
```

Comment:
hmm i wonder whether we can reuse the temporary directory that was already created to store key/digest/etc, and use a subdirectory of it as gpghome; eg in `verify_playbook()`:
```python
        gpg_home = pathlib.Path(temp_dir.name) / "gpghome"
        result: GPGCommandResult = verify_gpg_signed_file(gpg_home, digest_file, signature_file, key_file)
```

Reply:
I wouldn't mix those up. I'd like to treat `crypto.py` as independent mechanism shielded from the rest of the code. It will make maintenance (and/or debugging) easier, I believe.